### PR TITLE
Add demo content support

### PR DESCRIPTION
**Pull request by ssmith353**
Hey @marco-leo,
This needs some symantic versioning updates. Not sure what your systems for that are.

### CODE REVIEW

File: commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CommerceDiscountsImporter.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+				_assetCategoryLocalService.dynamicQuery();
+
+			Criterion criterion = RestrictionsFactoryUtil.eq(
+				"companyId", serviceContext.getCompanyId());
+
+			List<AssetCategory> assetCategories =
+				_assetCategoryLocalService.dynamicQuery(
+					dynamicQuery.add(criterion));
+
+			for (int i = 0; i < jsonArray.length(); i++) {
```
Comment (marco-leo) [inline commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CommerceDiscountsImporter.java:112]: @ssmith353 I would prefer a different way, where we do load a category per time, using _assetCategoryLocalService.fetchCategory , so we do avoid an internal loop

File: commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/JournalArticleImporter.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Steven Smith
+ */
+@Component(service = JournalArticleImporter.class)
+public class JournalArticleImporter {
+
+	public void importJournalArticles(
```
Comment (marco-leo) [inline commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/JournalArticleImporter.java:40]: @ssmith353 we do have a complex logic already i place, where we do check and import structures and templates, is still using an old pattern and we should migrate to the importer utils one, but we can just use this as in BrecciaInitializer: CPFileImporterImpl

**Review by marco-leo — CHANGES_REQUESTED** [review]

File: commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CommerceDiscountsImporter.java (hunk 1 above)
Comment (ssmith353) [reply commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CommerceDiscountsImporter.java:112]: Hey @marco-leo, 
assetCategoryLocalService does not have a way to fetch the category by the name. 

```fetchCategory(long groupId, long parentCategoryId, String name, long vocabularyId)```

```
fetchCategory(long categoryId)
```

Comment (ssmith353) [reply commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CommerceDiscountsImporter.java:112]: Hey @marco-leo Do you have any other feedback for this?

**Comment by marco-leo** [comment]
merged. Thanks

**Pull request closed without merging**